File: Swift/Day10(21.07.27)/SQLite/SQLite/Students.swift
//
//  Students.swift
//  SQLite
//
//  Created by RayAri on 2021/07/27.
//

import Foundation

class Students{
    var id: Int
    var name: String?
    var dept: String?
    var phone: String?
    
    init(id: Int, name: String?, dept: String?, phone: String?){
        self.id = id
        self.name = name
        self.dept = dept
        self.phone = phone
    }
}

File: Flutter/Day10(21.08.27)/sqlite/lib/students.dart
class Students {
  final int? id;
  final String code;
  final String name;
  final String dept;
  final String phone;

  Students(
      {this.id,
      required this.code,
      required this.name,
      required this.dept,
      required this.phone});

  //값을 가져올떄
  Students.fromMap(Map<String, dynamic> res)
      : id = res['id'],
        code = res['code'],
        name = res['name'],
        dept = res['dept'],
        phone = res['phone'];

  //값을 줄때
  Map<String, Object?> toMap() {
    return {'id': id, 'code': code, 'name': name, 'dept': dept, 'phone': phone};
  }
}
